jax/scipy/stats/geom.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import scipy.stats as osp_stats
+
+from ... import lax
+from ...numpy import lax_numpy as jnp
+from ...numpy._util import _wraps
+from ..special import xlogy, xlog1py
+
+@_wraps(osp_stats.geom.logpmf, update_doc=False)
+def logpmf(k, p, loc=0):
+    k, p, loc = jnp._promote_args_inexact("geom.logpmf", k, p, loc)
+    zero = jnp._constant_like(k, 0)
+    one = jnp._constant_like(k, 1)
+    x = lax.sub(k, loc)
+    log_probs = xlog1py(lax.sub(x, one), -p) + lax.log(p)
+    return jnp.where(lax.le(x, zero), -jnp.inf, log_probs)
```

Comment:
I think you might need the "double where" trick to avoid breaking the gradient at zero:

```python
invalid = lax.le(x, zero)
x = jnp.where(invalid, 1, x) 
log_probs = xlog1py(lax.sub(x, one), -p) + lax.log(p)
return jnp.where(invalid, -jnp.inf, log_probs)
```

it would be worth checking, though.

Reply:
Hi,
I might be interpreting something wrong here. Please correct me if I am wrong. I tried the following things

First with no change in code
```
g = lsp_stats.geom.logpmf
grad_g = jax.grad(g)
print(grad_g(0.0, 0.5)
```

and then same thing with change suggested above
for first case I got, ```DeviceArray(-0., dtype=float32)``` and for second I got ```DeviceArray(0, dtype=float32)```

Reply:
is this what we wanted to check?

Reply:
Yes, thanks – it appears that the change is not necessary then.